hack/tests/e2e-go-new.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# remove running containers on exit
+function cleanup() {
+  kind delete cluster
+}
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source ./hack/lib/common.sh
+source ./hack/lib/test_lib.sh
+
+tmp_root=/tmp
+tmp_sdk_root=$tmp_root/operator-sdk
+test_dir=./test
+tests=$test_dir/e2e-new
+
+export TRACE=1
+export GO111MODULE=on
+
+: ${K8S_VERSION:?"must be set"}
+
+prepare_staging_dir $tmp_sdk_root
+fetch_tools $tmp_sdk_root
+setup_envs $tmp_sdk_root
```

Comment:
I might be wrong but from what I can see we're spinning up a Kind cluster at `K8S_VERSION = v1.18.2` (passed from the Makefile) and then that's what `kubectl` will automatically point to in the e2e test itself.

We're not actually using `envtest` anywhere in the e2e suite itself to point to the existing Kind cluster:
```Go
testEnv = &envtest.Environment{
...
}
cfg, err := testEnv.Start()
```

So why do we have `setup_envs` right now if we're not using it?

I realize we probably want to keep the first pass simple and maybe later switch over to using envtest to run against a local control plane (or just point to the Kind cluster).
But for now can we comment this out. 
And also add a comment to mark this as a TODO for later.

Otherwise it's slightly confusing and makes it seem like we're already using envtest when we're not.

Reply:
Commands that run `make` will run unit tests, which require envtest tools.

Reply:
It is true.

Reply:
Ah got it. It's for the unit tests that get called in the sample operator itself.
Never needed to do it locally so totally forgot it's using envtest via with the default control-plane binary locations e.g `/usr/local/kubebuilder/bin/etcd`.

Reply:
Maybe a comment here would help clarify that these envtest vars are intended for the default unit-tests within the test-operator itself, and not for running our e2e test via envtest.

Just so that this script is easier to maintain in the future.